ui/src/frontend/globals.ts

```diff
@@ -337,7 +341,16 @@ class Globals {
   }
 
   set state(state: State) {
+    state = assertExists(state);
+    let readyStateSet = state.engine?.ready && !this._state?.engine?.ready;
     this._state = assertExists(state);
+    if(readyStateSet){
```

Comment:
Using assertExists() to not have to return undefined? Yes.  I probably didn't need to assert the state twice tho.  The rest of it was added, because I needed to emit something when the engine on the state changes to ready.

Reply:
I just mean the lack of spaces in 'if(readyStateSet){'

Reply:
No.  This happened because if I save normally in the Perfetto UI code while in Sokatoa then the formatting from Sokatoa effects Perfetto UI rather than using Perfetto UI's formatting, causing large file changes.

Reply:
Well that sucks.  :)  Anyway, it looked weird to me, so I pointed it out.  If it matches the Perfetto style then fine.  If it doesn't, please update it so they don't complain later when we try to upstream our changes.

Reply:
I still see that which doesn't seem to match our code style or the perfetto code style, so I assume they will ask you to update it if/when we try to upstream it.

Reply:
Does the Perfetto project not have some linter tool configured? If so, why is it not working for us? And if not, why not?

Reply:
To solve this, we could have prettier ignore the perfetto folder since it seems that Perfetto has it as a submodule, but never seems to set up prettier.

Reply:
It also seems like eslint is getting a conflict between our eslint vs perfetto's eslint setup
```
[Error - 11:36:41 AM] Error: Plugin "@typescript-eslint" was conflicted between "perfetto\ui\.eslintrc.js" and ".eslintrc.js » ./configs/base.eslintrc.js".
    at mergePlugins (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:598:19)
    at createConfig (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:701:9)
    at ConfigArray.extractConfig (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:877:33)
    at CLIEngine.isPathIgnored (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\eslint\lib\cli-engine\cli-engine.js:990:18)
    at CLIEngine.executeOnText (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\eslint\lib\cli-engine\cli-engine.js:899:38)
    at ESLint.lintText (C:\Users\a.levans\Desktop\sokatoa\perfetto\ui\node_modules\eslint\lib\eslint\eslint.js:593:23)
    at c:\Users\a.levans\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:24860
    at E (c:\Users\a.levans\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:19218)
    at e.validate (c:\Users\a.levans\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:24819)
    at c:\Users\a.levans\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:221494
```